WalletWasabi.Gui/Controls/WalletExplorer/CoinListView.xaml

```diff
@@ -74,31 +74,27 @@
       </StackPanel>
       <controls:BusyIndicator IsBusy="{Binding IsCoinListLoading}" Text="Loading...">
         <controls:ExtendedListBox Items="{Binding Coins}" VirtualizationMode="None" SelectedItem="{Binding Path=SelectedCoin, Mode=TwoWay}">
-          <controls:ExtendedListBox.ContextMenu>
-            <ContextMenu>
-              <!--Enqueuing needs password, TODO: jump to password box OR display pw box in context menu + send button-->
-              <MenuItem Command="{Binding EnqueueCoin}" IsVisible="false">
-                <MenuItem.Header>
-                  <Grid>
-                    <Grid.ColumnDefinitions>
-                      <ColumnDefinition />
-                      <ColumnDefinition />
-                    </Grid.ColumnDefinitions>
-                    <TextBlock Text="Enqueue Coin, password:" VerticalAlignment="Center" />
-                    <TextBox Width="50" Grid.Column="1" />
-                  </Grid>
-                </MenuItem.Header>
-              </MenuItem>
-              <MenuItem Header="Dequeue from CoinJoin" Command="{Binding DequeueCoin}">
-                <MenuItem.Icon>
-                  <DrawingPresenter HorizontalAlignment="Left" Height="16" Width="16" Stretch="Fill" Drawing="{StaticResource CoinListView_DequeueCoin}" />
-                </MenuItem.Icon>
-              </MenuItem>
-            </ContextMenu>
-          </controls:ExtendedListBox.ContextMenu>
           <controls:ExtendedListBox.ItemTemplate>
             <DataTemplate>
               <Grid>
+                <Grid.ContextMenu>
+                  <ContextMenu>
+                    <!--Enqueuing needs password, TODO: jump to password box OR display pw box in context menu + send button-->
+                    <MenuItem Command="{Binding $parent[controls:ExtendedListBox].DataContext.EnqueueCoin}" IsVisible="false">
```

Comment:
Isn't this a hack? @danwalmsley is this OK?

Reply:
its not ok!

@jmacato if you moved the commands to the `ItemsViewModel` then you dont need this!

Reply:
If you need to call a method on the PageViewModel for whatever reason then you need to pass that reference to the `ItemsViewModel`...

like `public ItemsViewModel(PageViewModel parent)`

we shouldnt capture those relationships in XAML (breaks the 1 to 1 relationship)

Reply:
Yes, I remember that time ago I asked about this pattern in the Avalonia gitter and you told @danwalmsley told me that that means something was wrong with mi model. 

I saw this during the revision but the alternative is to move code (properties and commands) to the right view models:

* AddressViewModel: (`IsHardwareWallet`, `ToggleQrCode`, `SaveQRCodeCommand`, `CopyAddress`, `CopyLabel`,`ChangeLabelCommand`)

* TransactionViewModel: `CopyTransactionId`

* CoinViewModel: (`DequeueCoin` and `EnqueueCoin`)

This refactoring is not free of risks imo and that's why I decided to approve it. It is not time to move chucks of code from here to there, in fact, I have tried to move as less code as possible during this code freeze.

Reply:
If we dont move the commands, to the corresponding VMs then we shouldnt  move the XAML either, otherwise we are building ourselves a mess.

We need to maintain a 1 to 1 relationship with Views and Viewmodels, all the commands and properties of a viewmodel must correspond to bindings in the xaml.

Reply:
Wait a moment. Why do we have the enqueue coin code there at all? We don't enqueue with context menu.